### PR TITLE
check existing setup before deployment

### PR DESCRIPTION
**Pull request by raghavendra-talur**
Signed-off-by: Raghavendra Talur <rtalur@redhat.com>

---
This change is [Reviewable](https://reviewable.io/reviews/gluster/gluster-kubernetes/175)

### CODE REVIEW

File: deploy/gk-deploy [hunk 1]
```diff
@@ -384,6 +384,11 @@ if [[ ${ABORT} -eq 1 ]]; then
 fi
 
 if [[ ${LOAD} -eq 0 ]]; then
+  check_pods "glusterfs=heketi-pod"
```
Comment (MohamedAshiqrh) [inline deploy/gk-deploy]: @raghavendra-talur My Bad. Check pods will abort if it didn't find the pods with these labels. 
I think we can add another flag which can parse in check pods and return appropriate. It is upto you.

Comment (raghavendra-talur) [inline deploy/gk-deploy]: Done. Thanks.

File: deploy/gk-deploy [hunk 2]
```diff
@@ -384,6 +389,10 @@ if [[ ${ABORT} -eq 1 ]]; then
 fi
 
 if [[ ${LOAD} -eq 0 ]]; then
+  if [[ $(are_pods_running "glusterfs=heketi-pod") == "yes" ]]; then
```
Comment (MohamedAshiqrh) [inline deploy/gk-deploy]: We can use True or false?

Comment (MohamedAshiqrh) [inline deploy/gk-deploy]: We can ignore for now.

**Comment by MohamedAshiqrh** [comment]
@humblec review and merge.

**Comment by humblec** [comment]
LGTM.

**Comment by jarrpa** [comment]
Why are we even modifying check_pods to return a string? If I don't hear back on this by tomorrow I'm reverting it.

**Comment by raghavendra-talur** [comment]
@jarrpa Earlier, check_pods was not returning anything. If pods are found, control flows back to caller and it proceeds as everything is good. If pods are not found, check_pods called abort() which lead to an exit.

IMO, check_pods is a util function which not change control flow of the program. Hence the change to make it return value to indicate if pods are running or not. The action to be taken is left to the caller.

If the question is why "string" instead of a int, it is just a preference that I have. We can change it to int or bool if you strongly feel so.

**Comment by jarrpa** [comment]
@raghavendra-talur Okay, agreed on the control flow point. However in testing I found other concerns, in particular that output is no longer properly displayed when running the command in a subshell. I'm submitting an alternative implementation now.

**Pull request merged**